### ISSUE TITLE
Make crashes

### ISSUE THREAD

**Issue by stevertaylor** [JohannesBuchner/PolyChord#2]
My attempts to run "make" always fail with the following error. I have set compiler type to be gnu.
   
    InitSampler.f90:7.8:

    use ini_module,               only: read_params,initialise_program
        1
    Fatal Error: Can't open module file 'ini_module.mod' for reading at (1): No such file or directory
    make[1]: *** [InitSampler.o] Error 1
    make: *** [libchord.a] Error 2

**Comment by JohannesBuchner** [comment]
Please get the official upstream version instead for the moment, the version here is out of date and only important if you want to run PyPolyChord (are you?). I am not sure what is the cause of that error, you might be running make from the wrong directory or have old files lying around.

**Comment by stevertaylor** [comment]
Ideally I would like to run pypolychord. Should I have the .mod file in question, or is it supposed to be made somewhere along the line?

On 12 Aug 2015, at 18:49, Johannes Buchner <notifications@github.com<mailto:notifications@github.com>> wrote:


Please get the official upstream version instead for the moment, the version here is out of date and only important if you want to run PyPolyChord (are you?). I am not sure what is the cause of that error, you might be running make from the wrong directory or have old files lying around.

**Comment by JohannesBuchner** [comment]
It should be built from src/ini.f90

Should like like this: 
```
user@computer ~/Downloads/PolyChord $ make COMPILER_TYPE=gnu MPI=which: no ifort in (/usr/local/bin:/usr/bin:/bin:/opt/bin:/usr/x86_64-pc-linux-gnu/gcc-bin/4.8.4:/usr/games/bin:/home/user/bin)
cd ./src && make libchord.a
make[1]: Entering directory '/mnt/data/daten/Master/home/Downloads/PolyChord/src'
gfortran -ffree-line-length-none -cpp  -Ofast -c utils.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c params.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c settings.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c abort.F90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c array_utils.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c priors.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c calculate.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c mpi_utils.F90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c random_utils.F90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c run_time_info.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c read_write.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c ini.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c chordal_sampling.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c clustering.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c feedback.f90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c generate.F90 -lstdc++  
gfortran -ffree-line-length-none -cpp  -Ofast -c nested_sampling.F90 -lstdc++  
ar r libchord.a priors.o read_write.o ini.o mpi_utils.o chordal_sampling.o calculate.o clustering.o generate.o abort.o random_utils.o run_time_info.o utils.o array_utils.o nested_sampling.o params.o settings.o feedback.o
ar: creating libchord.a
make[1]: Leaving directory '/mnt/data/daten/Master/home/Downloads/PolyChord/src'
```

**Comment by JohannesBuchner** [comment]
Ah I remember now and can reproduce the bug. In the Makefile, the order is not correctly reflecting the dependencies. A workaround is to run make with --keep-going and run it several times.

**Comment by stevertaylor** [comment]
Thanks for the tip. That gets me past that error, after which I now encounter the following:

    polychord.F90:152.33:

        output_info = NestedSampling(loglikelihood,priors,settings,0) 
                                     1
    Error: Interface mismatch in dummy procedure 'loglikelihood' at (1): Type mismatch in argument 'ndims' (INTEGER(4)/REAL(8))

**Comment by stevertaylor** [comment]
It seems this doesn't matter if I just want to use PyPolychord. With a few iterations of "make --keep-going" and "make libchord.so" it is able to give me libchord.so, which I then changed to a .dylib since I'm on a mac. I added a few lines to the top of run.py in pypolychord (similar to what you have in your pymultinest run.py file) to append a libchord extension based on system architecture. PyPolychord now seems to work well. Thanks!
